### ISSUE TITLE
Make all popovers implement a modal behavior

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#56077]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/56042#issuecomment-1807695604 

Turns out most of the popovers in the editor UI do implement constrained tabbing but some don't. I'm not sure what the reasoning behing this inconsistent is, but I'd tend to think all the popovers should implement a consistent keyboard interaction, regardless whether the component is 'inline' within the post content or in other parts of the UI.

### Step-by-step reproduction instructions

- Edit a post, select some text and press COmmand+K to add a link to the selected text.
- A popover open, focus is moved to the link input field.
- Press the Tab key multiple times
- Observe that tabbing is constrained within the popover.
- Add a link and save.
- Select the linked text.
- A popover with the Edit link UI appears.
- Press the Tab key to move within the popover.
- Move focus to the Edit button (the one with the pencil icon) and press Enter.
- A popover with the Edti link UI appears.
- Press the Tab key multiple times.
- Observe that tabbing is _not_ constrained within the popover.

There may be other instances of popovers where tabbing is not constrained. To me, it seems there is no good reason why popovers should not constrain tabbing. The keyboard interaction should always be the same: predictable and consistent. I'd like to hear thoughts also from @alexstine and @joedolson 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by alexstine** [comment]
@afercia I tend to agree but also note that some popovers don't take focus by default. The `AutocompleteUI` for example.

CC: @getdave.

**Comment by afercia** [comment]
Thanks @alexstine

**Comment by getdave** [comment]
I started looking into dialogs on Link Control here https://github.com/WordPress/gutenberg/pull/54063 although it's clearly a work in progress and should be considered with that in mind.

**Comment by mdrovdahl** [comment]
I think related...the tab behavior in the Link Control edit mode within the Patterns context is suboptimal. Additionally, this Link Control behavior in the Patterns context is inconsistent with same control in the post editor context.

Steps to reproduce:

1. Create a new site via Playground (https://playground.wordpress.net)
2. Navigate to: Appearance > Editor > Patterns > Add New Pattern
3. Within the default paragraph block, create and save some hyperlinked text
4. Edit the hyperlink, change the text
5. Press tab twice and the dialog closes without saving the changes from step 4


https://github.com/user-attachments/assets/6dc16e79-0008-471b-9fb2-94b61dbcae7a
